lib/vector-core/src/config/mod.rs

```diff
@@ -339,6 +340,31 @@ impl LogNamespace {
         }
     }
 
+    /// Vector: The `ingest_timestamp`, and `source_type` fields are added to "event metadata", nested
+    /// under the name "vector". This data will be marked as read-only in VRL.
+    ///
+    /// Legacy: The values of `source_type_key`, and `timestamp_key` are stored as keys on the event root,
+    /// only if a field with that name doesn't already exist.
```

Comment:
I might be totally missing something obvious... but why is the doc comment discussing Vector and Legacy, but appears (maybe that is my misunderstanding) , to only be doing stuff with the Vector case ?
If it's truly both then maybe a one-liner comment above each call in the function delineating which is Vector and which is Legacy would be helpful.

Reply:
`insert_standard_vector_source_metadata()` calls into `insert_vector_metadata()` twice, once for each of the current "vector metadata" for sources (`ingest_timestamp` and `source_type`).

```rust
        self.insert_vector_metadata(
            log,
            path!(log_schema().source_type_key()),
            path!("source_type"),
            Bytes::from_static(source_name.as_bytes()),
        );
        self.insert_vector_metadata(
            log,
            path!(log_schema().timestamp_key()),
            path!("ingest_timestamp"),
            now,
        );
```

The signature is `insert_vector_metadata(&self, log, legacy_key, metadata_key, value)`. So this function is inserting the name into either `%vector.source_type` or `source_type` (default value from [log_schema](https://vector.dev/docs/reference/configuration/global-options/#log_schema)) and the current time into either `%vector.ingest_timestamp` or `timestamp` (again, default).

After writing this out, I think the misunderstanding came from the name of the function. `insert_vector_metadata` doesn't refer to the `LogNamespace:Vector` but to the body of metadata `vector` as opposed to `source` (the type of the source).

Reply:
Ah, thanks for that explanation!
I guess the primary source of confusion for me was that the comment clearly differentiates between "Vector" and "Legacy" but it wasn't clear that the `insert_vector_metadata` was the one handling that.
I guess from a perspective of someone not super familiar with the log namespacing stuff, it was a bit confusing :sweat_smile:

Reply:
Yeah, that's a good note. @fuchsnj IDK if we want to think of a new name or just mark this as a "thing" to know.

Reply:
Yea, it's an unfortunate overlap in the naming between `Legacy` / `Vector` namespaces, and `source` / `vector` metadata.